test/integration/ContributionReward.spec.ts

```diff
@@ -0,0 +1,243 @@
+const Avatar = require('@daostack/arc/build/contracts/Avatar.json');
+const UController = require('@daostack/arc/build/contracts/UController.json');
+const AbsoluteVote = require('@daostack/arc/build/contracts/AbsoluteVote.json')
+
+describe('ContributionReward', () => {
+    let web3, addresses, opts, contributionReward;
+    beforeAll(async () => {
+        web3 = await getWeb3();
+        addresses = getContractAddresses();
+        opts = await getOptions(web3);
+        contributionReward = new web3.eth.Contract(ContributionReward.abi, addresses.ContributionReward, opts);
+    });
+
+    it('Sanity', async () => {
+        const accounts = web3.eth.accounts.wallet;
+
+        // START long setup ...
+        const daoToken = await new web3.eth.Contract(DAOToken.abi, undefined, opts)
+            .deploy({ data: DAOToken.bytecode, arguments: ['Test Token', 'TST', '10000000000'] })
+            .send();
+
+        const reputation = await new web3.eth.Contract(Reputation.abi, undefined, opts)
+            .deploy({ data: Reputation.bytecode, arguments: [] })
+            .send();
+        await reputation.methods.mint(accounts[1].address, 100000).send(); // to be able to pass a vote
+
+        const avatar = await new web3.eth.Contract(Avatar.abi, undefined, opts)
+            .deploy({ data: Avatar.bytecode, arguments: ['Test', daoToken.options.address, reputation.options.address] })
+            .send();
+
+        const controller = await new web3.eth.Contract(UController.abi, undefined, opts)
+            .deploy({ data: UController.bytecode, arguments: [] })
+            .send();
+
+        const absVote = await new web3.eth.Contract(AbsoluteVote.abi, undefined, opts)
+            .deploy({ data: AbsoluteVote.bytecode, arguments: [] })
+            .send();
+
+        const setParams = absVote.methods.setParameters(20, true);
+        const absVoteParamsHash = await setParams.call()
+        await setParams.send()
+        const crSetParams = contributionReward.methods.setParameters(0, absVoteParamsHash, absVote.options.address);
+        const paramsHash = await crSetParams.call();
+        await crSetParams.send()
+
+        await avatar.methods.transferOwnership(controller.options.address).send();
+        await controller.methods.newOrganization(avatar.options.address).send();
+        await controller.methods.registerScheme(
+            contributionReward.options.address,
+            paramsHash,
+            '0x0000001F', // full permissions,
+            avatar.options.address
+        ).send();
+        // END setup
+
+        const descHash = '0x0000000000000000000000000000000000000000000000000000000000000123';
+        const rewards = {
+            rep: 1,
+            nativeToken: 2,
+            externalToken: 3,
+            eth: 4,
+            periods: 5,
+            periodLength: 6
+        }
+        const propose = contributionReward.methods.proposeContributionReward(
```

Comment:
need await ?

Reply:
no, that's only building the object which you can `call(..)` or `send(..)` on.
Below we do both and `await` the results.